core/templates/dev/head/domain/objects/NumberWithUnitsObjectFactory.js

```diff
@@ -126,11 +126,20 @@ oppia.factory('NumberWithUnitsObjectFactory', [
     };
 
     NumberWithUnits.fromDict = function(numberWithUnitsDict) {
-      return new NumberWithUnits(
-        numberWithUnitsDict.type,
-        numberWithUnitsDict.real,
-        numberWithUnitsDict.fraction,
-        numberWithUnitsDict.units);
+      if (numberWithUnitsDict.type === 'fraction') {
```

Comment:
It would be better to do this check in the constructor function rather than here.

Reply:
We need to check whether the input is `fraction` or not. Then only we can convert it to fraction dict else it will throw errors.

Reply:
Yes but you can do that in the constructor function, right?

Reply:
You can do something like this:
```
if (type == 'fraction') {
  fraction = FractionObjectFactory.fromDict(fraction);
}

unit = UnitsObjectFactory(unit);

return NumberWithUnits(.....);